### ISSUE TITLE
dell-recovery shows no error when the target disk space is insufficient

### ISSUE THREAD

**Issue by alex-tu-cc** [dell/dell-recovery#82]
There's a feature request to warn people that the space of target storage is not enough.

[LP: #1842325 dell-recovery shows no error when the target disk space is insufficient](https://bugs.launchpad.net/oem-priority/+bug/1842325)

**Comment by superm1** [comment]
I think the hardest part of this is going to be coming up with an accurate calculation of how much storage space is needed.  At install time you will be able to determine the size of the squashfs, and can look at the `Installed-Size` attribute of stuff in debs/main, but there are usually lot of other packages that are installed using modaliases.  Also if any of the packages installed replace versions from the squashfs the calculations need to compare the `Installed-Size` of the one in squashfs with one packaged in a deb.

The other interesting aspect from that bug is that swapfile.  The policy to avoid making swap partitions obviously doesn't apply to swap files, so perhaps ubiquity needs a way to turn off swap file creation.  It would allow smaller disks to be safely used I think like this.

So in short; The "easier" solution however is to just filter smaller disks from the UX and from being offered.

**Comment by superm1** [comment]
Ah found this:
```
d-i partman-swapfile/percentage string 5
d-i partman-swapfile/size string 2048
```

> Setting either of those to zero, will result in system without any swap at all. And one can tweak relative integer percentage points and absolute limits in integer percentage points or MiB.


https://blog.surgut.co.uk/2016/12/swapfiles-by-default-in-ubuntu.html
